### PR TITLE
[REG-546] master [fix] display recently used vendors on the top of the list on product

### PR DESCRIPTION
**Pull request by romadjan**

### CODE REVIEW

File: addons/consumption_agreement/models/consumption_agreement.py [hunk 1]
```diff
@@ -120,6 +120,21 @@ def _compute_possible_partners(self):
...
+                if line.vendor_id not in line.product_id.seller_ids.mapped('partner_id'):
+                    self.env['product.supplierinfo'].create([
+                        {
+                            'partner_id': line.vendor_id.id,
+                            'product_tmpl_id': line.product_id.product_tmpl_id.id,
+                            'price': 0
+                        }
+                    ])
+                new_last_vendor = line.product_id.seller_ids.filtered(lambda seller: seller.partner_id == line.vendor_id)
+                new_last_vendor.sequence = 0
```
Comment (chupaSV) [inline addons/consumption_agreement/models/consumption_agreement.py:137]: Why don't you set it in create?

File: addons/regency_estimate/models/purchase_order.py [hunk 2]
```diff
@@ -132,6 +132,51 @@ def button_cancel(self):
...
+
+        # Add the partner in the supplier list of the product if the supplier is not registered for
+        # this product. We limit to 10 the number of suppliers for a product to avoid the mess that
+        # could be caused for some generic products ("Miscellaneous").
+        for line in self.order_line:
+            # Start custom logic
+            sequence = 1
+            for seller in line.product_id.seller_ids.sorted('sequence'):
+                seller.sequence = sequence
+                sequence += 1
```
Comment (chupaSV) [inline addons/regency_estimate/models/purchase_order.py:146]: Use call of super() after the sequence above, instead of copy

**Review by chupaSV — CHANGES_REQUESTED** [review]

File: addons/regency_estimate/models/purchase_order.py [hunk 3]
```diff
@@ -132,6 +132,14 @@ def button_cancel(self):
                 }
         return super(PurchaseOrder, self).button_cancel()
 
+    def _add_supplier_to_product(self):
+        for line in self.order_line:
+            line.product_id._recompute_sequence()
+        super(PurchaseOrder, self)._add_supplier_to_product()
+        for line in self.order_line:
+            partner = self.partner_id if not self.partner_id.parent_id else self.partner_id.parent_id
```
Comment (chupaSV) [inline addons/regency_estimate/models/purchase_order.py:140]: This can be done before the loop

**Review by chupaSV — CHANGES_REQUESTED** [review]

File: addons/regency_estimate/models/purchase_order.py (hunk 3 above)
Comment (romadjan) [reply addons/regency_estimate/models/purchase_order.py]: done

**Review by chupaSV — APPROVED** [review]

**Pull request merged**